src/game/common/rts/team.cpp

```diff
@@ -1492,3 +1496,721 @@ bool Team::Damage_Team_Members(float amount)
+            if (!Loco_Set_Matches(update->Get_Locomotor_Set()->Get_Valid_Surfaces(), surfaces)) {
+                it.Advance();
+                continue;
+            }
+        } else if (!Loco_Set_Matches(1, surfaces)) {
+            it.Advance();
+            continue;
+        }
+
+        if (!it.Cur()->Is_Effectively_Dead()) {
+            if (!it.Cur()->Is_KindOf(KINDOF_INERT)) {
+                if (it.Cur()->Did_Exit(poly)) {
+                    b2 = true;
+                } else {
+                    if (it.Cur()->Is_Inside(poly)) {
+                        b3 = true;
+                    }
+                }
+
+                b1 = true;
+            }
+        }
+
+        it.Advance();
+    }
+
+    return b1 && b2 && !b3;
+}
+
+bool Team::All_Inside(PolygonTrigger *poly, unsigned int surfaces) const
+{
+    if (!m_enteredOrExited) {
+        return false;
+    }
+
+    bool b1 = false;
+    bool b2 = false;
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
+        AIUpdateInterface *update = it.Cur()->Get_AI_Update_Interface();
+
+        if (update != nullptr) {
+            if (!Loco_Set_Matches(update->Get_Locomotor_Set()->Get_Valid_Surfaces(), surfaces)) {
+                it.Advance();
+                continue;
+            }
+        } else if (!Loco_Set_Matches(1, surfaces)) {
+            it.Advance();
+            continue;
+        }
+
+        if (!it.Cur()->Is_Effectively_Dead()) {
+            if (!it.Cur()->Is_KindOf(KINDOF_INERT)) {
+                if (!it.Cur()->Is_Inside(poly)) {
+                    b2 = true;
+                }
+
+                b1 = true;
+            }
+        }
+
+        it.Advance();
+    }
+
+    return b1 && !b2;
+}
+
+bool Team::None_Inside(PolygonTrigger *poly, unsigned int surfaces) const
+{
+    if (!m_enteredOrExited) {
+        return false;
+    }
+
+    bool b1 = false;
+    bool b2 = false;
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
+        AIUpdateInterface *update = it.Cur()->Get_AI_Update_Interface();
+
+        if (update != nullptr) {
+            if (!Loco_Set_Matches(update->Get_Locomotor_Set()->Get_Valid_Surfaces(), surfaces)) {
+                it.Advance();
+                continue;
+            }
+        } else if (!Loco_Set_Matches(1, surfaces)) {
+            it.Advance();
+            continue;
+        }
+
+        if (!it.Cur()->Is_Effectively_Dead()) {
+            if (!it.Cur()->Is_KindOf(KINDOF_INERT)) {
+                if (it.Cur()->Is_Inside(poly)) {
+                    b2 = true;
+                }
+
+                b1 = true;
+            }
+        }
+
+        it.Advance();
+    }
+
+    return b1 && !b2;
+}
+
+bool Team::Some_Inside_Some_Outside(PolygonTrigger *poly, unsigned int surfaces) const
+{
+    bool b1 = false;
+    bool b2 = false;
+    bool b3 = false;
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
```

Comment:
Such loop can be simplified as

```c++
for (; !it.Done(); it.Advance())
```

This way `it.Advance();` can be omitted from various places within the loop.

Reply:
Want to keep things matching pseudo to make it easier to spot issues.